best-practices/require-drop-cap-net-raw/require-drop-cap-net-raw.yaml

```diff
@@ -25,19 +26,19 @@ spec:
         - resources:
             kinds:
               - Pod
-      preconditions:
-        all:
-        - key: "{{ request.operation || 'BACKGROUND' }}"
-          operator: NotEquals
-          value: DELETE
+            operations:
+            - CREATE
+            - UPDATE
       validate:
-        message: >-
-          Containers must drop the `CAP_NET_RAW` capability.
-        foreach:
-          - list: request.object.spec.[ephemeralContainers, initContainers, containers][]
-            deny:
-              conditions:
-                all:
-                - key: CAP_NET_RAW
-                  operator: AnyNotIn
-                  value: "{{ element.securityContext.capabilities.drop[].to_upper(@) || `[]` }}"
\ No newline at end of file
+        cel:
+          variables:
+            - name: mustDropCapabilities
+              expression: "['CAP_NET_RAW','NET_RAW']"
+            - name: allContainers
+              expression: "object.spec.containers + object.spec.?initContainers.orValue([]) + object.spec.?ephemeralContainers.orValue([])"
+          expressions:
+            - expression: >-
+                variables.allContainers.all(container, 
+                container.?securityContext.?capabilities.?drop.orValue([]).exists(capability, capability.upperAscii() in variables.mustDropCapabilities))
+              message: >-
+                Containers must drop the `CAP_NET_RAW` capability.
```

Comment:
Why do we use the `validate.cel` subrule here? We already have the best practices written in CEL expressions.

I expect to solve the bug using the existing foreach.

Reply:
I picked up the wrong policy file. I should have picked following one -> https://github.com/kyverno/policies/blob/main/best-practices-cel/require-drop-cap-net-raw/require-drop-cap-net-raw.yaml
will revert the changes and update the correct policy file